library/common/src/main/java/com/google/android/exoplayer2/MediaCodecParameters.java

```diff
@@ -0,0 +1,44 @@
+package com.google.android.exoplayer2;
+
+import android.os.Build;
+import androidx.annotation.Nullable;
+import java.util.HashMap;
+
+/**
+ * MediaCodec parameters configuring the {@link android.media.MediaCodec} instances.
+ *
+ * once instantiated, the application can add key-value pairs calling {@link MediaCodecParameters#set}
+ * and send a message of type {@code Renderer#MSG_SET_CODEC_PARAMETERS} to the renderers.
+ *
+ */
+public class MediaCodecParameters {
+
+  public static final String KEY_DIALOG_ENHANCEMENT = "dialog-enhancement-gain";

```

Comment:
How confident are we that this generalizes properly to other codecs? As soon as some other codec manufacturer wants 5 levels rather than 3, wouldn't that be a problem? I kinda feel that in absence of standardization at the platform level, we should require apps wanting to use codec specific features to use them without too much abstraction. E.g.,

`codecParameters.setInt(DOLBY_DIALOG_ENHANCEMENT_GAIN, 3);`

This also removes the need for Dolby specific code in `MediaCodecAudioRenderer`. Open to discussion on this point though!

Reply:
Hi, 

IMO, going for a generic "dialog-enhancement-gain" also mean a "common" range.

We have proposed here 3 "levels" (while AC4 actually built-in Dialog Enhancement gain ranges from 0 to 12 dB and the gain we could apply on (e)AC3 content using our audio-effect toolbox ranges from 0 to 9 )  primarily to let the listener clearly hear the difference between 2 steps....  
If we allow a 1:1 mapping between the codec actual range and the application, then we essentially delegate to the application developer the understanding that this dialog-enhancement can go from [0,12] for AC4 , [0,9] for AC3, eAC3, potentially [0,4] for some other parties codec ... 

For a "codec" specific range, my take would be that the app developer would use this setCodecParameter API with the codec specific key directly... then indeed we would have to add the business logic to bundle the unmanaged parameter(s) and pass them directly to the codec. With Android12 onwards, exoplayer can now check which parameter is supported by the codec and filter only these params that can be passed through to the codec directly....